### PR TITLE
Modified the return statement for ADD call | Added tests to main.py and graph_memory.py

### PR DESCRIPTION
**Pull request by prateekchhikara**
## Description

Modified the return statement for ADD call, now memories (or updates) are returned instead of `{"message" : "ok"}`

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by Dev-Khant — APPROVED** [review]

**Comment by prateekchhikara** [comment]
The add call now supports a different format:

```
{
    'results': [
               {'memory': "Friend's name is John", 'event': 'ADD'},
      ],
    'relations': [
              {'source': 'alice123', 'relationship': 'friend', 'target': 'john'}
      ]
}
```

The get_all and search supports the following format:

```
{
      'results':[
               {'memory': "Friend's name is John", 'event': 'ADD'},
               {'memory': 'John has a dog named Tommy', 'event': 'ADD'}
      ],
      'relations': [
              {'source': 'alice123', 'relationship': 'friend', 'target': 'john'},
              {'source': 'john', 'relationship': 'has_dog', 'target': 'tommy'}
      ]
}
```

File: mem0/memory/graph_memory.py [hunk 1]
```diff
@@ -140,8 +149,15 @@ def _search(self, query, filters):
...
-                relation_list.extend(item['arguments']['relations'])
+                try:
+                    node_list.extend(item['arguments']['nodes'])
+                except Exception as e:
+                    logger.error(f"Error in search tool: {e}")
+
+                # try:
+                #     relation_list.extend(item['arguments']['relations'])
+                # except Exception as e:
+                #     logger.error(f"Error in search tool: {e}")
```
Comment (deshraj) [inline mem0/memory/graph_memory.py:160]: Can we please remove this since already commented?

File: mem0/memory/main.py [hunk 2]
```diff
@@ -18,7 +18,10 @@
 from mem0.memory.storage import SQLiteManager
 from mem0.memory.telemetry import capture_event
 from mem0.memory.utils import get_fact_retrieval_messages, parse_messages
-from mem0.utils.factory import EmbedderFactory, LlmFactory, VectorStoreFactory
+from mem0.configs.prompts import get_update_memory_messages
+from mem0.utils.factory import LlmFactory, EmbedderFactory, VectorStoreFactory
+from mem0.configs.base import MemoryItem, MemoryConfig
+import concurrent
```
Comment (deshraj) [inline mem0/memory/main.py:24]: Can you please run the lint command?

File: tests/test_main.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+    memory_instance.vector_store.list.assert_called_once_with(filters={"user_id": "test_user"}, limit=100)
+    
+    if enable_graph:
+        memory_instance.graph.get_all.assert_called_once_with({"user_id": "test_user"})
+    else:
+        memory_instance.graph.get_all.assert_not_called()
+
+if __name__ == "__main__":
+    pytest.main()
```
Comment (deshraj) [inline tests/test_main.py:222]: Dont need this line I think.

**Review by deshraj — APPROVED** [review]
Left minor comment else looks good.

**Pull request merged**